### DISCUSSION TITLE
Download Mac version

### DISCUSSION THREAD

**Discussion by floZx** [foldynl/QLog#130] [General]
Hello, is it possible to have a mac version for download?

**Comment by foldynl** [comment]
Hi,

unfortunately, there is no official DMG package. It is a long story about the support of Mac. A short story is that I have no Apple's HW to test it.

What I currently know:
1) Qlog can be compiled on Mac
2) Internally, I have DMG package prepared by GitHub but I know that the package is not working now. Unfortunately, I don't know why.

If you have any experience you could help me to debug it.

**Comment by floZx** [comment]
Hi,
I can try to provide a dmg but when compiling I get an error: Failure to find: res/flags/flags.qrc
I don't have any experience with QT but I can try to understand.

**Comment by foldynl** [comment]
You have to clone the QLog repo by command below because QLog repo contains a git-submodule (flags)

```git clone --recurse-submodules https://github.com/foldynl/QLog.git```

It would be great if you could help me with this. I've added a DMG file that GITHub creates as a result of Git Action.

You can find 
 - the zip file with DMG - https://github.com/foldynl/QLog/releases/download/v0.15.0/qlog.zip (it is reported that it immediately crashes)
-  GitHub Action for DMG - https://github.com/foldynl/QLog/blob/master/.github/workflows/macOSBuild.yml.

**Comment by floZx** [comment]
The app error in GITHub DMG is: 

- Library missing","details":["(terminated at launch; ignore backtrace)"],"reasons":["Library not loaded: @rpath\/QtDBus.framework\/Versions\/A\/QtDBus"

I don't know git action but apparently QTDBus is missing.

Compile on my iMac now work. I will do some tests and I can try to provide  DMG (Intel and M1) to make them available.

**Reply by foldynl** [reply]
> Compile on my iMac now work. I will do some tests and I can try to provide DMG (Intel and M1) to make them available.

It will be great. The problem is that I make a release approximately once a month. It will be necessary to figure out how to release packages without your assistance. It is a need to automate the release process therefore I use GitHub Actions. Please, are you able to review why the Action generate invalid packages?

**Comment by foldynl** [comment]
Maybe this is the solution - [source QT homepage](https://doc.qt.io/qt-6/macos-issues.html)

_D-Bus and macOS_

_The [QtDBus](https://doc.qt.io/qt-6/qtdbus-module.html) module defaults to dynamically loading the libdbus-1 library on macOS. That means applications linking against the [QtDBus](https://doc.qt.io/qt-6/qtdbus-module.html) module will load even on macOS systems that do not have the libraries, but they will fail to connect to any D-Bus server and they will fail to open a server using [QDBusServer](https://doc.qt.io/qt-6/qdbusserver.html)._

_To use D-Bus functionality, you need to install the libdbus-1 library, for example through Homebrew, Fink or MacPorts. You may want to include those libraries in your application's bundle if you're deploying to other systems. Additionally, note that there is no system bus on macOS and that the session bus will only be started after launchd is configured to manage it._

**Comment by floZx** [comment]
You may need to add this to the script:
`brew install dbus-glib` and `cp /usr/local/lib/libdbus-1.dylib qlog.app/Contents/Frameworks/libdbus-1.dylib`

**Comment by foldynl** [comment]
please, could you test a new package?

https://github.com/foldynl/QLog/releases/download/v0.15.0/qlog_new.zip

**Comment by floZx** [comment]
Unfortunately there is still a problem with QtDBus

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/QtDBus.framework/Versions/A/QtDBus
Referenced from: <90E7BB34-066C-32BC-BFD8-3D99769D4B76> /Applications/qlog.app/Contents/Frameworks/QtGui.framework/Versions/A/QtGui
Reason: tried: '/Applications/qlog.app/Contents/Frameworks/QtGui.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Applications/qlog.app/Contents/Frameworks/QtOpenGL.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Applications/qlog.app/Contents/Frameworks/QtOpenGLWidgets.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Applications/qlog.app/Contents/Frameworks/QtCharts.framework/Versions/A/../../../QtDBus.framework/Versions/A/QtDBus' (no such file), '/Applications/qlog.app/Contents/Frameworks/QtDBus.framework/Versions/A/QtDBus' (no such file), '/Applications/qlog.app/Contents/Frameworks/QtDBus.framework/Versions/A/QtDBus' (no such file), '/System/Volumes/Preboot/Cryptexes/OS@rpath/QtDBus.framework/Versions/A/Qt
(terminated at launch; ignore backtrace)
```

**Comment by foldynl** [comment]
that's bad news. I'll try to think of something, but I'm afraid that I won't be able to do anything with it without proper HW (or help).

**Comment by foldynl** [comment]
QtDBus has been added - to be honest, really don't know why it should be added because QLog does not use it. Maybe it is dependency of some other library under Qt6

please, could you try a new build ? 

https://github.com/foldynl/QLog/releases/download/v0.15.0/qlog_new2.zip

**Reply by floZx** [reply]
Ok, now it's ok for QtDBus. New missing library:
```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/QtQuickWidgets.framework/Versions/A/QtQuickWidgets
```

**Comment by foldynl** [comment]
it really doesn't make sense to me why I should use this module. QLog doesn't call anything like that. I think that some QT libraries are prepared differently than under Windows or Linux.

A new DMG is being prepared. It should be in 20 minutes.  I will  send you another zip...

Do you know is there some way to find out what all is missing there?

**Comment by foldynl** [comment]
thanks for your patience...another zip file

https://github.com/foldynl/QLog/releases/download/v0.15.0/qlog_new3.zip

**Reply by floZx** [reply]
For this build:
```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @loader_path/libbrotlicommon.1.dylib
Referenced from: <AC73DD9B-29B2-37C7-8212-669533064C26> /Applications/qlog.app/Contents/Frameworks/libbrotlidec.1.dylib
Reason: tried: '/Applications/qlog.app/Contents/Frameworks/libbrotlicommon.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS@loader_path/libbrotlicommon.1.dylib' (no such file), '/Applications/qlog.app/Contents/Frameworks/libbrotlicommon.1.dylib' (no such file), '/usr/local/lib/libbrotlicommon.1.dylib' (no such file), '/usr/lib/libbrotlicommon.1.dylib' (no such file, not in dyld cache)
(terminated at launch; ignore backtrace)
```

**Reply by floZx** [reply]
Other missing librarie:
- libicui18n.71 
and it seems to work... 🤞

**Comment by foldynl** [comment]
another attempt

https://github.com/foldynl/QLog/releases/download/v0.15.0/qlog_new4.zip

to be honest, , I don't understand at all what I'm doing (fixing) now

**Reply by floZx** [reply]
The app is working now. 
Good job.

**Comment by foldynl** [comment]
thanks for the confirmation and for patience.I was kind of losing hope that it would ever work. Just for my curiosity - are you running M1/M2 or Intel?

**Reply by floZx** [reply]
I tried the app on Intel and M1 (with Rosetta).

**Comment by foldynl** [comment]
Perfect...thank you.
